osu.Game.Rulesets.Catch/Skinning/Default/CatchHitObjectPiece.cs

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using JetBrains.Annotations;
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics.Containers;
+using osu.Game.Rulesets.Catch.Objects.Drawables;
+using osu.Game.Rulesets.Objects.Drawables;
+using osuTK.Graphics;
+
+namespace osu.Game.Rulesets.Catch.Skinning.Default
+{
+    public class CatchHitObjectPiece : CompositeDrawable
+    {
+        public readonly Bindable<Color4> AccentColour = new Bindable<Color4>();
+        public readonly Bindable<bool> HyperDash = new Bindable<bool>();
+
+        [Resolved(canBeNull: true)]
+        [CanBeNull]
+        protected DrawableHitObject DrawableHitObject { get; private set; }
+
+        [CanBeNull]
+        protected BorderPiece BorderPiece;
```

Comment:
Imo this is pretty weird and we've avoided doing this elsewhere in the game, but I'll overlook it here. At least it's properly annotated.

@peppy are you fine with this?

Reply:
As in having this at a base class even though it's never instantiated here? I guess we'd usually have a `CreateBorderPiece` method, could probably add that in?

Reply:
I agree it is kind of weird.
But I couldn't do `CreateBorderPiece` because I wanted to make set all `InternalChildren` on the derived class' constructor.

Reply:
I imagined it was something like that... Your recent changes improve it enough for me, so I'll resolve this.